pandas/core/indexes/base.py

```diff
@@ -6687,7 +6711,12 @@ def get_slice_bound(self, label, side: Literal["left", "right"]) -> int:
             else:
                 return slc
 
-    def slice_locs(self, start=None, end=None, step=None) -> tuple[int, int]:
+    def slice_locs(
+        self,
+        start: slice_type = None,
+        end: slice_type = None,
```

Comment:
slice_type could just be Hashable. I believe we require that all columns and indices are hashable.